### ISSUE TITLE
`baseBranch` issue with multiple dev branches

### ISSUE THREAD

**Issue by maccapakka** [atlassian/changesets#533]
## Problem

We're implementing Changesets into our company monorepo and coming across a problem due to our branching strategy. 

We use;
`@next`:  to stock up on breaking changes 
`@current`: minor and patch changes - basically master 
`@previous`: legacy support 

Since the config is `.json` we cannot dynamically set the `baseBranch` property 

When developing in `@next` we have to regularly merge `@current` to keep up with any feature requests/fixes

And likewise when `@next` is ready to release this gets merged into and becomes `@current`

You can imagine during the merge conflicts we have to keep cherry-picking this `baseBranch` property and making sure it's the correct value depending on which branch you're on 

Is there a way around this?

Thanks in advance!

**Comment by Andarist** [comment]
Hmm, I would assume that after resolving conflicts correctly for the first time this shouldn't happen often. Git only applies patches when merging and tries to resolve conflicts between them and the code that they are applied to. So if your patches (what you merge) don't contain any changes to that `.json` file it should just be OK with it.

I would have to give this a deeper thought - not sure how we could design those options in a way that would alleviate that pain for you. If you have any ideas - please share them so we can discuss them further.

**Comment by rohit-gohri** [comment]
If the config file is a `js` file then one can conditionally set the base branch according to current branch?

**Comment by Andarist** [comment]
Technically yes, but having different config formats tends to be problematic. OTOH our config is quite slim and Changesets is not invoked often so that's not that big of a concern.

However, if `.js` configs would solve your problem you can quite easily just write a simple script that would alter the existing config, invoke Changesets and roll back your changes. Before introducing new features I'd like to first propose alternatives and observe how popular of a request this is.

**Comment by rohit-gohri** [comment]
Makes sense 👍. Can be done through `package.json` scripts I think, something like this works for me in a quick test:

```json
"scripts": {
	"prechangeset": "node scripts/generateConfig.js",
    "changeset": "changeset"
}
```

And instead of `npx changeset add` you can then do `npm run changeset -- add`

**Comment by Andarist** [comment]
Yep, either this or just a short script that would exec changeset under the hood - so you would have the whole thing within a single npm script.
